### PR TITLE
Add X-Api-Key middleware

### PR DESCRIPTION
**Pull request by miguel-botelho**
This PR adds a simple authenticator middleware that checks if the `X-Api-Key´ received is the same as the one set in the environment variable.

### CODE REVIEW

File: cmd/api/api.go [hunk 1]
```diff
@@ -17,6 +18,20 @@ func main() {
 	handler.Logger.Fatal(handler.Start(":" + port))
 }
 
+func apiKeyMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		fmt.Println("HELLO")
+		headers := c.Request().Header
+		key := headers.Get("X-Api-Key")
+		fmt.Println(key)
```
Comment (romeufcrosa) [inline cmd/api/api.go:26]: Debug?

Comment (miguel-botelho) [reply cmd/api/api.go:26]: fock

File: cmd/api/api.go [hunk 2]
```diff
@@ -17,6 +18,20 @@ func main() {
 	handler.Logger.Fatal(handler.Start(":" + port))
 }
 
+func apiKeyMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		fmt.Println("HELLO")
```
Comment (romeufcrosa) [inline cmd/api/api.go:23]: Is it me you're looking for?

File: cmd/api/api.go (hunk 1 above)
Comment (miguel-botelho) [reply cmd/api/api.go]: You saw nothing.

**Review by romeufcrosa — APPROVED** [review]
Damn, you fast!

**Pull request merged**